packages/embark-webserver/src/index.js

```diff
@@ -78,30 +82,32 @@ class WebServer {
         this.setServiceCheck();
       });
     });
-  }
-
-  setServiceCheck() {
-    const self = this;
-
-    this.events.request("services:register", 'Webserver', function (cb) {
-      let url = self.protocol + '://' + canonicalHost(self.host) + ':' + self.port;
-      checkIsAvailable(url, function (available) {
-        let devServer = __('Webserver') + ' (' + url + ')';
-        let serverStatus = (available ? 'on' : 'off');
-        return cb({name: devServer, status: serverStatus});
-      });
-    });
 
     this.events.on('check:wentOffline:Webserver', () => {
       this.logger.info(__("Webserver is offline"));
     });
+    this.events.on('check:backOnline:Webserver', () => {
+      this.logger.info(__("Webserver is online"));
+    });
+  }
+
+  setServiceCheck() {
+    if (!this.isServiceRegistered) {
+      this.isServiceRegistered = true;
+      this.events.request("services:register", 'Webserver', (cb) => {
+        let url = this.protocol + '://' + canonicalHost(this.host) + ':' + this.port;
+        checkIsAvailable(url, function (available) {
+          let devServer = __('Webserver') + ' (' + url + ')';
+          let serverStatus = (available ? 'on' : 'off');
+          return cb({name: devServer, status: serverStatus});
+        });
+      });
+    }
   }
 
   listenToCommands() {
     this.events.setCommandHandler('placeholder:build', (cb) => this.buildPlaceholderPage(cb));
     this.events.setCommandHandler('browser:open', (cb) => this.openBrowser(cb));
-    this.events.setCommandHandler('webserver:start', (cb) => this.events.request('processes:launch', 'webserver', cb));
-    this.events.setCommandHandler('webserver:stop',  (cb) => this.events.request('processes:stop', 'webserver', cb));
```

Comment:
This is a breaking change no? Even though we give out a warning about this command being deprecated, we should keep it for at least one major version cycle?

Reply:
I don't think users ever knew about or used the `webserver:start` events. We can keep it until the next major, but not sure it makes a difference

Reply:
The *command* is still there, but deprecated (with message), however I simply removed the *event handler*, which I wouldn't think that anyone would be using. However, you're right in that if anyone was using this event, it would be a breaking change. I can remove it if we think it's the best approach.

Reply:
Updated.